crates/saya/core/src/dojo_os/piltover.rs

```diff
@@ -21,23 +22,35 @@ pub async fn starknet_apply_piltover(
     calldata: PiltoverCalldata,
     contract: Felt,
     account: &SayaStarknetAccount,
-    nonce: Felt,
+    mut nonce: Felt,
```

Comment:
As we mentioned, here we shouldn't have to use the nonce explicitly. Even if we want to batch transactions, starknet doesn't have yet nonce channels for instance.

But @chudkowsky you mentioned you are on a rework and let's definitely do this in an other PR. But we shouldn't need to specify the nonce, and let `starknet-rs` managing it for us.
Or may I overlooked something?